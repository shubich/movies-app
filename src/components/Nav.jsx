import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const Nav = ({ search = true }) => (
  <div id="nav">
    <h1 className="sitename">netflixroulette</h1>
    {
      search ? (
        <Link to="/">
          <button className="btn btn-default btn-search">SEARCH</button>
        </Link>
      ) : null
    }
  </div>
);

Nav.propTypes = {
  search: PropTypes.bool.isRequired,
};

export default Nav;
