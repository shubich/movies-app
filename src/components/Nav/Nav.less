@import "../../styles/palette";

.Nav {
    display: flex;
    flex-flow: row wrap;
    box-sizing: border-box;
}

.Nav-SiteName {
    flex: 1;
}
