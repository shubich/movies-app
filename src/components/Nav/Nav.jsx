import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import Button from '../Button';
import SiteName from '../SiteName';

import './Nav.less';

const Nav = ({ search }) => (
  <div className="Nav">
    <SiteName className="Nav-SiteName" />
    {
      search ? (
        <Link to="/">
          <Button search>SEARCH</Button>
        </Link>
      ) : null
    }
  </div>
);

Nav.propTypes = {
  search: PropTypes.bool,
};

Nav.defaultProps = {
  search: true,
};

export default Nav;
