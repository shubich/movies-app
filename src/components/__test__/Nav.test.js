import React from 'react';
import { Nav } from '../Nav';
import { shallow } from 'enzyme';

let props;

describe('Nav', () => {
    it('renders correctly with search', () => {
        const wrapper = shallow(
            <Nav/>
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly without search', () => {
        const wrapper = shallow(
            <Nav search={false} />
        );
        expect(wrapper).toMatchSnapshot();
    });
})