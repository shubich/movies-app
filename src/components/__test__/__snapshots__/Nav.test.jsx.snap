// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav renders correctly with search 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Nav
    search={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="sitename"
        >
          netflixroulette
        </h1>,
        <Link
          replace={false}
          to="/"
        >
          <button
            className="btn btn-default btn-search"
          >
            SEARCH
          </button>
        </Link>,
      ],
      "id": "nav",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "netflixroulette",
          "className": "sitename",
        },
        "ref": null,
        "rendered": "netflixroulette",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": <button
            className="btn btn-default btn-search"
          >
            SEARCH
          </button>,
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "SEARCH",
            "className": "btn btn-default btn-search",
          },
          "ref": null,
          "rendered": "SEARCH",
          "type": "button",
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="sitename"
          >
            netflixroulette
          </h1>,
          <Link
            replace={false}
            to="/"
          >
            <button
              className="btn btn-default btn-search"
            >
              SEARCH
            </button>
          </Link>,
        ],
        "id": "nav",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "netflixroulette",
            "className": "sitename",
          },
          "ref": null,
          "rendered": "netflixroulette",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": <button
              className="btn btn-default btn-search"
            >
              SEARCH
            </button>,
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "SEARCH",
              "className": "btn btn-default btn-search",
            },
            "ref": null,
            "rendered": "SEARCH",
            "type": "button",
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Nav renders correctly without search 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Nav
    search={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="sitename"
        >
          netflixroulette
        </h1>,
        null,
      ],
      "id": "nav",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "netflixroulette",
          "className": "sitename",
        },
        "ref": null,
        "rendered": "netflixroulette",
        "type": "h1",
      },
      null,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="sitename"
          >
            netflixroulette
          </h1>,
          null,
        ],
        "id": "nav",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "netflixroulette",
            "className": "sitename",
          },
          "ref": null,
          "rendered": "netflixroulette",
          "type": "h1",
        },
        null,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
